src/azure-cli/azure/cli/command_modules/storage/_validators.py

```diff
@@ -117,7 +102,9 @@ def validate_client_parameters(cmd, namespace):
         if not n.account_name:
             n.account_name = get_config_value(cmd, 'storage', 'account', None)
         if auth_mode == 'login':
-            n.token_credential = _create_token_credential(cmd.cli_ctx)
+            from azure.cli.core._profile import Profile
+            profile = Profile(cli_ctx=cmd.cli_ctx)
+            n.token_credential, _, _ = profile.get_login_credentials(resource="https://storage.azure.com")
```

Comment:
Do we want to hard-code `resource`? Shouldn't it be added to the cloud? Just want to confirm.

Reply:
it is hard-coded resource, not related to cloud.